Source/SwiftLintFramework/Models/ConfigurationError.swift

```diff
@@ -0,0 +1,13 @@
+//
+//  ConfigurationError.swift
+//  SwiftLint
+//
+//  Created by Scott Hoyt on 1/19/16.
+//  Copyright © 2016 Realm. All rights reserved.
+//
+
+import Foundation
+
+public enum ConfigurationError: ErrorType {
```

Comment:
This was something I thought about for a bit. Basically what was the best way to support configuration structures failing and provide optional configuration values functionality. To use the failable initializers approach that was previously used in `ConfigurableRule` would have required each rule to have its own `RuleConfig` type because you could only declare the default values in the structure. The second approach was to split initialization with default values and configuration with optional values into `init` and `setConfig`. Since we needed a way to indicate failure in `setConfig`, we could either throw or return some sort of success/failure flag (e.g. Bool, `Result`, etc.). Throwing felt the most Swifty to me, and I needed something to throw, so that's how this came to be.

There's certainly a case to be made that if we use this approach, we should introduce some finer-grain errors, possibly merge this with YamlParsingError, and probably change `Configuration.init?` to `Configuration.init() throws`. That's all possible in this PR, but in my head I was scoping it to the rework that needs to be done on logging and errors.